sssd_test_framework/utils/sssd.py

```diff
@@ -1039,3 +1039,28 @@ def pam(self, features: list[str] | None = None) -> None:
 
         self.sssd.authselect.select("sssd", features)
         self.sssd.enable_responder("pam")
+
+    def socket_responders(self, responders: list[str] | None = None) -> None:
+        """
+        Configure SSSD for socket-activated responders.
+
+        Sets an empty services line in sssd.conf to enable socket activation
+        for specified responders.
+
+        :param responders: List of responders to enable via socket activation.
+                          If None, enables all known socket responders.
+        :type responders: list[str] | None
+        :raises RuntimeError: If enabling or starting a socket unit fails.
+        """
+        if responders is None:
+            responders = ["nss", "pam", "sudo", "ssh", "pac", "autofs"]
+
+        # Set empty services to enable socket activation
+        self.sssd.sssd["services"] = ""
+
+        # Ensure socket units are enabled and started
+        for responder in responders:
+            socket_unit = f"sssd-{responder}.socket"
+
+            # Start the socket unit (raises on error by default)
+            self.sssd.host.conn.run(f"systemctl start {socket_unit}")
```

Comment:
This should use the svc utility, it will make sure the change is reverted before starting other test and it will also print all available logs on an error.